tests/external_executor/executors/wiki_cacher.py

```diff
@@ -115,7 +116,8 @@ def run_loop(self, activated_event):
 
             for work in stub.Activate(Empty()):
                 if work.HasField("activated"):
-                    activated_event.set()
+                    if activated_event is not None:
```

Comment:
I wonder if we should remove this event altogether and rely on an informational endpoint on the CCF node to list the external executors currently registered on that node? For now, we can simply wait for the registered endpoints to not return a 404? @DomAyre @eddyashton @MahatiC what do you think?

Reply:
An endpoint that lists the active executors could also work, but it involves another polling loop. I think maintaining this push/event style notification is nice, even if we decide our sample infra is polling rather than using it? Its plausible that an executor will want to do some set up at exactly this point (after it has activated, before it processes any user requests).

Waiting for the registered endpoint to not return a 404 is less precise - when we have multiple executors serving the same URI this won't work.

Reply:
We need to expose from CCF that an executor is ready to handle requests, even for operation purposes, as well as testing.

Reply:
Done it by polling the registered executor endpoints for now https://github.com/microsoft/CCF/pull/5018/commits/90ae68730b9d7c2d62236d3188a7098bcedcfc82 but I agree that a proper endpoint to check for registered executors should follow this PR

Reply:
@DomAyre may be worth ticketing so we don't forget :)